whylog/log_reader/searchers.py

```diff
@@ -68,3 +84,11 @@ def _reverse_from_offset(self, offset, buf_size=BufsizeConsts.STANDARD_BUF_SIZE)
                         yield line, actual_offset
             actual_offset = self._decrease_actual_offset_properly(actual_offset, truncated)
             yield truncated, actual_offset
+
+    def search(self, investigation_step):
```

Comment:
this method should search in N, not 2N. By default, starting offset should not be acquired with so high cost, but it should be provided by FrontInput which has offset of the effect line. But it is not used.